src/GafferScene/SceneAlgo.cpp

```diff
@@ -389,27 +391,30 @@ uint64_t g_contextUniquefierValue = 0;
 
 SceneAlgo::History::Ptr historyWalk( const CapturedProcess *process, InternedString scenePlugChildName, SceneAlgo::History *parent )
 {
-	SceneAlgo::History::Ptr history;
-	ScenePlug *scene = const_cast<Plug *>( process->plug.get() )->parent<ScenePlug>();
-	if( scene && process->plug.get() == scene->getChild( scenePlugChildName ) )
+	SceneAlgo::History::Ptr result;
+	Plug *plug = const_cast<Plug *>( process->destinationPlug.get() );
+	while( plug )
```

Comment:
From a maintainability perspective, it might be nice to have a little friendly into here as to the relationship between the iteration of the plugs upstream of the destination plug to the computed plug, and the children of `process`. Once you have the picture in your head it makes sense...

Reply:
97f9e5a.

Reply:
Sorry to be totally boring on this one. The logical 'click' when I was reviewing this originally was that:

- The captured process tree is a hierarchy of discreet processes that actually computed values in order to provide the final value of the requested plug, which may include computation for other plugs such as internal plugs, dependencies or expression referenced plugs - as an explanation for the need to double check names/parent.

- Each process knows the ultimate destination plug, but not anything about the intermediate hops via connections - hence the need for the two stage iteration.